test/unittest/tests/block/test_block_other.cpp

```diff
@@ -19,27 +19,23 @@
 static void
 test_block_header_byteswap_copy(WT_BLOCK_HEADER *from, WT_BLOCK_HEADER *to)
 {
-    WT_BLOCK_HEADER prev_from, prev_to;
+    WT_BLOCK_HEADER prev_from;
 
     // Save the original values before any potential byte re-orderings.
     prev_from.disk_size = from->disk_size;
     prev_from.checksum = from->checksum;
-    prev_to.disk_size = to->disk_size;
-    prev_to.checksum = to->checksum;
 
-#ifdef WORDS_BIGENDIAN
     __wt_block_header_byteswap_copy(from, to);
+
+#ifdef WORDS_BIGENDIAN
     REQUIRE(to->checksum == __wt_bswap32(prev_from.checksum));
     REQUIRE(to->disk_size == __wt_bswap32(prev_from.disk_size));
-#else
-    // Test that the byte orderings remain the same in both block headers.
```

Comment:
These checks didn't make sense in the original test. In little endian mode we would assign `prev_from = from` and `prev_to = to` and then assert they're identical which is trivially true.

Checking that `prev_to == to` also didn't make sense to me. `to` is the destination buffer so why would we expect it to be unchanged? We also have one case where we call this function in-place (`test_block_header_byteswap_copy(&to, &to)`) so I've moved the validation up into the callers where it's clearer what we're trying to assert.